src/main/java/org/dependencytrack/persistence/VulnerabilityPolicyQueryManager.java

```diff
@@ -229,4 +257,97 @@ public int updateVulnerabilityPolicyByName(VulnerabilityPolicy vulnerabilityPoli
             }
         }
     }
+
+    private String buildOrderBySubQuery() {
+        String orderBySubQuery = null;
+        if (orderBy == null) {
+            orderBySubQuery =
+                    """
+                                ORDER BY "NAME" 
+                            """;
+        } else {
+            orderBySubQuery =
+                    "ORDER BY " + "\"" + mapFieldName(orderBy) + "\"";
+        }
+        if (orderDirection == OrderDirection.ASCENDING) {
+            orderBySubQuery += " ASC ";
+        } else if (orderDirection == OrderDirection.DESCENDING) {
+            orderBySubQuery += " DESC ";
+        }
+        orderBySubQuery += """
+                , "ID"
+                """;
+        return orderBySubQuery;
+    }
+
+    private static String mapFieldName(String fieldName) {
+        if (fieldName == null) {
+            return null;
+        }
+        return switch (fieldName.toUpperCase()) {
+            case "AUTHOR", "CREATED", "VALID_FROM", "VALID_UNTIL" -> fieldName.toUpperCase();
+            default -> throw new IllegalArgumentException("Ordering not supported");
+        };
+    }
+
+    private String addPaginationToQuery(final String query) {
+        if (pagination != null && pagination.isPaginated()) {
+            return query +
+                    """
+                                OFFSET %d
+                                LIMIT %d;
+                            """.formatted(pagination.getOffset(), pagination.getLimit());
+        }
+        return query;
+    }
+
+    private static VulnerabilityPolicy mapVulnerabilityPolicyResultSet(ResultSet rs) throws SQLException {
+        VulnerabilityPolicy vulnerabilityPolicy = new VulnerabilityPolicy();
+        vulnerabilityPolicy.setId(rs.getLong("ID"));
+        JSONObject analysisJson = new JSONObject(rs.getString("ANALYSIS"));
+        vulnerabilityPolicy.setAnalysis(MAPPER.convertValue(analysisJson, VulnerabilityPolicyAnalysis.class));
+        vulnerabilityPolicy.setAuthor(rs.getString("AUTHOR"));
+        Array conditions = rs.getArray("CONDITIONS");
+        vulnerabilityPolicy.setConditions((String[]) conditions.getArray());
+        vulnerabilityPolicy.setCreated(rs.getTimestamp("CREATED"));
+        vulnerabilityPolicy.setDescription(rs.getString("DESCRIPTION"));
+        vulnerabilityPolicy.setName(rs.getString("NAME"));
+        JSONArray ratingsJson = new JSONArray(rs.getString("RATINGS"));
+        vulnerabilityPolicy.setRatings(MAPPER.convertValue(ratingsJson, new TypeReference<>() {
+        }));
```

Comment:
I just stumbled over this in testing: There needs to be a `null` check here as `ratings` are optional.

Also I think you can pass the result of `rs.getString("RATINGS")` directly to `MAPPER.readValue`. No need to convert to `JSONArray` first.